frontend/src/components/auth/components/sign-up-form/sign-up-form.tsx

```diff
@@ -14,33 +18,79 @@ const SignUpForm: FC<Props> = ({ onSubmit }) => {
-        <p>
-          <Input
-            type="text"
-            label="Email"
-            placeholder="Enter your email"
-            name={getNameOf<UserSignUpRequestDto>('email')}
-            control={control}
-            errors={errors}
-          />
-        </p>
-        <p>
-          <Input
-            type="text"
-            label="Password"
-            placeholder="Enter your password"
-            name={getNameOf<UserSignUpRequestDto>('password')}
-            control={control}
-            errors={errors}
-          />
-        </p>
-        <Button type="submit" label="Sign up" />
-      </form>
+      <div>
+        <div className={styles.background}>
+          <div className={`${styles.circle} ${styles.circleFirst}`} />
+          <div className={`${styles.circle} ${styles.circleSecond}`}>
+            <img src={logo} className={styles.logo} />
+            <img src={authImage} />
+          </div>
+          <div className={`${styles.circle} ${styles.circleThird}`} />
+          <div className={`${styles.circle} ${styles.circleFourth}`} />
+        </div>
+        <div className={styles.formWrapper}>
+          <div className={styles.formHeader}>
+            <h2>Create an account</h2>
+            <p>
+              Already registered?{' '}
+              <Link to={AppRoute.SIGN_IN}>
+                <span>Log in</span>
+              </Link>
+            </p>
+          </div>
+          <form onSubmit={handleSubmit(onSubmit)}>
+            <div className={styles.formContent}>
+              <p className={styles.formInputWrapper}>
+                <Input
+                  type="text"
+                  label="Full Name"
+                  name={getNameOf<UserSignUpRequestDto>('fullName')}
+                  control={control}
+                  errors={errors}
+                  className={errors.fullName && styles.error}
+                />
+              </p>
+              <p className={styles.formInputWrapper}>
+                <Input
+                  type="text"
+                  label="Email"
+                  name={getNameOf<UserSignUpRequestDto>('email')}
+                  control={control}
+                  errors={errors}
+                  className={errors.email && styles.error}
+                />
+              </p>
+              <p className={styles.formInputWrapper}>
+                <Input
+                  type="password"
+                  label="Password"
+                  name={getNameOf<UserSignUpRequestDto>('password')}
+                  control={control}
+                  errors={errors}
+                  className={errors.password && styles.error}
+                />
+              </p>
+            </div>
+            <div className={styles.buttonWrapper}>
+              <Button
+                className={styles.formButton}
+                type="submit"
+                label="Sign up"
+                disabled={Object.keys(errors).length > 0}
```

Comment:
Or even declare memoized hasErrors flag with this check
```suggestion
                disabled={Boolean(Object.keys(errors).length)}
```